Pod/Classes/Highlightr.swift

```diff
@@ -236,26 +237,26 @@ open class Highlightr
         }
       }
       
-      scanner.scanLocation += 1
+      scanner.currentIndex = scanner.string.index(after: scanner.currentIndex)
       
-      let string = scanner.string as NSString
-      let nextChar = string.substring(with: NSMakeRange(scanner.scanLocation, 1))
-      if(nextChar == "s")
-      {
-        scanner.scanLocation += (spanStart as NSString).length
-        scanner.scanUpTo(spanStartClose, into:&scannedString)
-        scanner.scanLocation += (spanStartClose as NSString).length
-        propStack.append(scannedString! as String)
-      }
-      else if(nextChar == "/")
-      {
-        scanner.scanLocation += (spanEnd as NSString).length
+      let nextChar = scanner.string[scanner.currentIndex]
+      if(nextChar == "s") {
+        scanner.currentIndex = scanner.string.index(scanner.currentIndex, offsetBy: spanStart.utf8.count)
+        scannedString = scanner.scanUpToString(spanStartClose)
+        /// Test that advancing the index of the string is in the boundary of the string without invoking
+        /// `offsetBy` which is unsafe.
+        let currentString = scanner.string[scanner.string.startIndex..<scanner.currentIndex]
+        if currentString.utf8.count + spanStartClose.utf8.count < scanner.string.utf8.count {
```

Comment:
The fix is here. Check that the next possible scanner position is in the string range.